src/main/java/nextstep/optional/User.java

```diff
@@ -33,7 +35,11 @@ public static boolean ageIsInRange1(User user) {
     }
 
     public static boolean ageIsInRange2(User user) {
-        return false;
+        return Optional.ofNullable(user).filter(u -> u != null)
+            .map(u -> user.age)
+            .filter(age -> age != null)
+            .filter(age -> age >= 30 && age <= 45)
+            .isPresent();
```

Comment:
불필요한 필터 연산을 줄이고 이렇게도 쓸 수 있을거 같아요.
```suggestion
return Optional.ofNullable(user) 
        .map(User::getAge) 
        .filter(age -> age >= 30 && age <= 45)
        .isPresent();
```